lib/msf/ui/console/command_dispatcher/auxiliary.rb

```diff
@@ -51,35 +51,36 @@ def cmd_run(*args, action: nil)
       jobify = true
     end
 
-    rhosts = datastore['RHOSTS']
+    mod_with_opts = mod.replicant
```

Comment:
Given the references held in these complex objects, has anyone checked how well we're GCing all of this? I would wager a small amount to say that the more we have these patterns used, the more we're eating into our memory and runtime overhead. Probably worth profiling if not already being done in CI

Reply:
The last memory investigation work I did was here https://github.com/rapid7/metasploit-omnibus/pull/119#issuecomment-631791726, and omnibus now ships with jemalloc to help with memory fragmentation. I don't think this PR will exacerbate memory issues. The replicant pattern is needed as modules unfortunately mutate their own internal state, a decision from before my time.

Ruby 3.x will ship with have a new memory allocator/gc implementation too https://bugs.ruby-lang.org/issues/17570 / https://bugs.ruby-lang.org/issues/18045, so all bets are off on how this will all behave in the future 😄